### PR TITLE
feat: show number of problems solved on profile

### LINKED ISSUE #1083: [Feature]: Add number of problems solved
### Description

another suggestion: can add how many problems i have solved in profile. also how many hards/easy/medium division 

### Explanation

_No response_

### PR DESCRIPTION
**Pull request by zaCKoZAck0**
Added a UI to view number of challenges solved. 
## Description

Added
1. Total Challenges solved
2. Total Challenges
## Related Issue

Closes #1083 

## Motivation and Context

Tried to take motivation from multiple places, didn't find something useful. Made with suggestions from the community.

#### Light mode mobile/desktop
Screenshot from 2023-11-13 10-59-03
Screenshot from 2023-11-13 10-58-46
#### Dark mode mobile/desktop
Screenshot from 2023-11-13 10-58-37
Screenshot from 2023-11-13 10-57-42

### CODE REVIEW

File: apps/web/src/app/[locale]/(profile)/[username]/_components/dashboard/overview-tab.tsx [hunk 1]
```diff
@@ -13,18 +14,25 @@ export async function OverviewTab({ user }: Props) {
...
+        <div className="max-w-sm flex-grow" />
+      </div>
+      <Card className="flex-grow">
+        <CardHeader>
+          <CardTitle>Bio</CardTitle>
+        </CardHeader>
+        <CardContent>
+          <div className="w-full text-sm">
+            {Boolean(hasBio) && <Markdown>{user.bio}</Markdown>}
+            {Boolean(!hasBio) && (
```
Comment (jhechtf) [inline apps/web/src/app/[locale]/(profile)/[username]/_components/dashboard/overview-tab.tsx:30]: you shouldn't need the extra `Booleam` wrappers here.

Comment (zaCKoZAck0) [reply apps/web/src/app/[locale]/(profile)/[username]/_components/dashboard/overview-tab.tsx:30]: Not my part of code I think, anyways I'll fix. Thanks.

File: apps/web/src/app/[locale]/(profile)/[username]/_components/dashboard/_actions.ts [hunk 2]
```diff
@@ -64,3 +66,89 @@ export async function getChallengeHistoryByCategory(type: HistoryType, userId: s
...
+  });
+
+  const allChallenges = await prisma.challenge.groupBy({
+    by: ['difficulty'],
+    _count: {
+      _all: true,
+    },
+  });
+
+  const totalSolved = challengesSolved.reduce((acc, challenge) => acc + challenge._count._all, 0);
```
Comment (bautistaaa) [inline apps/web/src/app/[locale]/(profile)/[username]/_components/dashboard/_actions.ts:107]: am i crazy or is this just a `.length`

File: apps/web/src/app/[locale]/(profile)/[username]/_components/dashboard/_actions.ts [hunk 3]
```diff
@@ -64,3 +66,89 @@ export async function getChallengeHistoryByCategory(type: HistoryType, userId: s
...
+  const totalChallenges = allChallenges.reduce((acc, challenge) => acc + challenge._count._all, 0);
```
Comment (bautistaaa) [inline apps/web/src/app/[locale]/(profile)/[username]/_components/dashboard/_actions.ts:108]: am i crazy or is this just a `.length`

**Review by bautistaaa — APPROVED** [review]

**Pull request merged**